### ISSUE TITLE
[Bug] Transport name is empty in outbound transport request for TChannel channel transport

### ISSUE THREAD

**Issue by ChenX1993** [yarpc/yarpc-go#2293]
## Issue
The transport name (field: `Transport`) is empty in outbound [transport request ](https://github.com/yarpc/yarpc-go/blob/ed351c67ea70a8d0ec36c0cd218bb4f3826e5845/api/transport/request.go#L42) for TChannel channel transport.

## Reproduce
Run a unit test with debug mode:
1. Add a breaking point in `firstoutboundmiddleware`: 
https://github.com/yarpc/yarpc-go/blob/ed351c67ea70a8d0ec36c0cd218bb4f3826e5845/internal/firstoutboundmiddleware/middleware.go#L51
![Screenshot 2024-09-15 at 1 30 47 PM](https://github.com/user-attachments/assets/305361fe-f675-4a0e-ab6f-cda8b831b286)

2. Run a TChannel channel related unit test, e.g.
https://github.com/yarpc/yarpc-go/blob/ed351c67ea70a8d0ec36c0cd218bb4f3826e5845/transport/tracer_test.go#L195

And we can clearly see the `Transport` field of the transport request is empty.
![Screenshot 2024-09-15 at 1 33 12 PM](https://github.com/user-attachments/assets/750506e5-3bce-4e2e-97fe-02fd79843892)

### TLDR
The TChannel's [ChannelOutbound](https://github.com/yarpc/yarpc-go/blob/ed351c67ea70a8d0ec36c0cd218bb4f3826e5845/transport/tchannel/channel_outbound.go#L66) doesn't implement the [transport.Namer](https://github.com/yarpc/yarpc-go/blob/ed351c67ea70a8d0ec36c0cd218bb4f3826e5845/api/transport/outbound.go#L47) interface.

### Details
For outbound transport request, the transport name is set by the [firstoutboundmiddleware](https://github.com/yarpc/yarpc-go/blob/ed351c67ea70a8d0ec36c0cd218bb4f3826e5845/internal/firstoutboundmiddleware/middleware.go), and the firstoutboundmiddleware tries to cast the `transport.Outbound` interface to `transport.Namer` interface, and uses `TransportName()` method of the `transport.Namer` interface to get the transport name.
https://github.com/yarpc/yarpc-go/blob/ed351c67ea70a8d0ec36c0cd218bb4f3826e5845/internal/firstoutboundmiddleware/middleware.go#L66-L76

However, since the TChannel channel outbound doesn't implement `transport.Namer` interface, `firstoutboundmiddleware` hence cannot work for it.
